### ISSUE TITLE
Incorrect suggestions due to incorrect card selection.

### ISSUE THREAD

**Issue by n4ru** [corlettb/Arena-Helper#2]
This is more of an issue caused by the main fork of Arena-Helper and the way it detects what card was actually selected. A lot of picks are detected wrong by the plugin, causing incorrect data to be submitted to HearthArena, causing the entire draft to end up pretty awkward and flat out bad.

Could an option be added to force the HA suggestion to be submitted (highest scored card), rather than whatever the plugin detects?

**Comment by corlettb** [comment]
I did notice some bad detection when running at odd resolutions... like using 1920x1080 fullscreen when the desktop running much higher.

Having the desktop at 1080p and running the game at 1080p for me resulted in perfect detection. 

Its probably better to figure out what breaks the detection (might be variance in resolutions) and raise it on the main fork rather than implementing work arounds.

There is a debug option you can turn on in <home>\AppData\HearthstoneDeckTracker\ArenaHelper\arenahelper.json that can give more info while the detection is taking place.

**Comment by n4ru** [comment]
Will check the debug stuff now and post back here.

However, I am running at 1080p.

**Comment by corlettb** [comment]
Both desktop and game?

**Comment by n4ru** [comment]
Yep.

**Comment by corlettb** [comment]
Ahhh I see you have already raised an issue. https://github.com/rembound/Arena-Helper/issues/3

He suggests its down to plugin system and the 100ms update timer and the click not being picked up correctly. I think the code will default to the first card in the selection if it "missed" it.

Lame I know... but did you try holding the mouse button down for an extended period of time in place over each card?

If he is right, to fix the functionality he'd need to either change the way the detection is done. Difficult via OCR. Or the plugin system would have to change in HDT so they can pick up mouse click events.

**Comment by corlettb** [comment]
Maybe another lame fix would be the plugin to say "Opps I missed that... tell me what it was".

**Comment by n4ru** [comment]
I figured that wasn't an issue you could solve in your fork since it affects everything, which is why I suggested "defaulting" to the HA suggestion in your fork.

**Comment by n4ru** [comment]
In your fork it matters because it screws up the whole draft, not just the one card, since the incorrect pick is actually being submitted and modifies future choices.

Defaulting to HA's pick should fix this until a "proper" fix is implemented in the main plugin fork.

**Comment by corlettb** [comment]
Well it still matters in the main fork because the saved deck at the end would need to be corrected. We could "default" to the highest ranked card. But these are only suggestions. I don't always picked the card suggested.

I think it needs to be accurate and a fix that works for both versions.

**Comment by n4ru** [comment]
Of course, was merely a suggestion for a bandaid fix, not a permanent solution.

**Comment by corlettb** [comment]
Should I comment on the other issue and suggest a "Oops I missed that..." and present the choice to the user? I could possibly implement it... although I suspect rembound is far better than me at C# ui changes.

**Comment by n4ru** [comment]
I suppose so, but I'll probably try and hack together the HA defaulting myself in your fork.

Now I gotta go set up compilers :package:
